src/tools/ceph_objectstore_tool.cc

```diff
@@ -719,6 +719,74 @@ int ObjectStoreTool::export_files(ObjectStore *store, coll_t coll)
   return 0;
 }
 
+int set_inc_osdmap(ObjectStore *store, epoch_t e, bufferlist& bl) {
+  OSDMap::Incremental inc;
+  bufferlist::iterator it = bl.begin();
+  inc.decode(it);
+  if (e == 0) {
+    e = inc.epoch;
+  } else if (e != inc.epoch) {
+    cerr << "incremental.epoch mismatch: "
+	 << inc.epoch << " != " << e << ".\n"
+	 << "But will continue anyway." << std::endl;
+  }
```

Comment:
I think you should set inc.epoch = e here.  Also, maybe you want to utilize the existing --force flag to allow this but only if the --force set.